web/packages/shared/hooks/useAsync.ts

```diff
@@ -125,10 +126,10 @@ export function useAsync<Args extends unknown[], AttemptData>(
             return [null, new CanceledError()] as [AttemptData, Error];
           }
 
-          setState(prevState => ({
-            ...prevState,
+          setState(() => ({
             status: 'error',
-            statusText: err?.message,
+            error: err,
+            statusText: err && err['message'],
             data: null,
```

Comment:
This spread here was unnecessary as we overwrote all three possible fields anyway.